### PR TITLE
media/voice: Add Speech detector prototype and Software EPD solution

### PR DESCRIPTION
**Pull request by gcjjyy**
Add SpeechDetector prototype for co-work.
Most functions are not implemented yet.

Add Software EPD solution(Fixed float) at the External.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [65f638ad02e52fa3182b424505bd8b9fbd56a530] - Code Rule Check (C++) OK.

File: framework/src/media/voice/SpeechDetector.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+namespace media {
+namespace voice {
+
+static SpeechDetector *g_instance = nullptr;
+
+SpeechDetectorInterface *SpeechDetectorInterface::instance()
+{
+	if (!g_instance) {
+		g_instance = new SpeechDetector();
```
Comment (junmin-kim) [inline framework/src/media/voice/SpeechDetector.cpp]: To implement singleton, this way can be problem.
How about using `static SpeechDector instance`?

Comment (gcjjyy) [reply framework/src/media/voice/SpeechDetector.cpp]: You're right. I'll fix this!

**Comment by seinfra** [comment]
Target : [57f1f391349e968a459f5f66a3ed714467ff6aec] - Code Rule Check OK.

**Comment by gcjjyy** [comment]
@sunghan-chang You're right. I'll follow the rules!

**Pull request merged**